includes/Abilities/Excerpt_Generation/Excerpt_Generation.php

```diff
@@ -0,0 +1,273 @@
+<?php
+/**
+ * Excerpt generation WordPress Ability implementation.
+ *
+ * @package WordPress\AI
+ */
+
+declare( strict_types=1 );
+
+namespace WordPress\AI\Abilities\Excerpt_Generation;
+
+use WP_Error;
+use WordPress\AI\Abstracts\Abstract_Ability;
+use WordPress\AI_Client\AI_Client;
+
+use function WordPress\AI\get_post_context;
+use function WordPress\AI\get_preferred_models;
+use function WordPress\AI\normalize_content;
+
+/**
+ * Excerpt generation WordPress Ability.
+ *
+ * @since 0.1.0
+ */
+class Excerpt_Generation extends Abstract_Ability {
+
+	/**
+	 * The default number of candidates to generate.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var int
+	 */
+	protected const CANDIDATES_DEFAULT = 3;
+
+	/**
+	 * Returns the input schema of the ability.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return array<string, mixed> The input schema of the ability.
+	 */
+	protected function input_schema(): array {
+		return array(
+			'type'       => 'object',
+			'properties' => array(
+				'content'    => array(
+					'type'              => 'string',
+					'sanitize_callback' => 'sanitize_text_field',
+					'description'       => esc_html__( 'Content to generate excerpt suggestions for.', 'ai' ),
+				),
+				'post_id'    => array(
+					'type'              => 'integer',
+					'sanitize_callback' => 'absint',
+					'description'       => esc_html__( 'Content from this post will be used to generate excerpt suggestions. This overrides the content parameter if both are provided.', 'ai' ),
+				),
+				'candidates' => array(
+					'type'              => 'integer',
+					'minimum'           => 1,
+					'maximum'           => 10,
+					'default'           => self::CANDIDATES_DEFAULT,
+					'sanitize_callback' => 'absint',
+					'description'       => esc_html__( 'Number of excerpts to generate', 'ai' ),
+				),
+			),
+		);
+	}
+
+	/**
+	 * Returns the output schema of the ability.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return array<string, mixed> The output schema of the ability.
+	 */
+	protected function output_schema(): array {
+		return array(
+			'type'       => 'object',
+			'properties' => array(
+				'excerpts' => array(
```

Comment:
Since we're only going to return a single excerpt, I think we can change this schema to be simpler

Reply:
@dkotter This has been updated to return a string with the single excerpt.